src/descriptor/mod.rs

```diff
@@ -241,6 +241,40 @@ impl<Pk: MiniscriptKey> Descriptor<Pk> {
         Ok(Descriptor::Tr(Tr::new(key, script)?))
     }
 
+    /// For a Taproot descriptor, returns the internal key.
+    pub fn internal_key(&self) -> Option<&Pk> {
```

Comment:
Perhaps name it `tap_internal_key()` to keep it consistent with the other methods and to make it clearer that its taproot-related and taproot-only?

Reply:
The other methods yield a `TapTree` and are named as such, with no prefix. It's just more obvious that a "taptree" is a taproot object while an "internal key" is less obvious.

I can stick a `taproot_` prefix onto here if you want, though that makes the method name longer and won't make it look consistent with the other methods. Since Taproot is the "default" output type for the forseeable future, I would almost rather stick `non_taproot_` onto everything else..

Reply:
If Taproot is considered the 'default' then not having a prefix definitely makes sense. I didn't think of it as such.